### PR TITLE
issue-1512

### LINKED ISSUE #1512: Implement `TransformerSqlparser::transform_ddl`
Part of #1218

### PR DESCRIPTION
**Pull request by SGZW**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

solve #1512 

## Changelog

- New Feature

## Related Issues

Fixes #1512 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: query/src/sql/parser/transformer/transform_sqlparser.rs [hunk 1]
```diff
@@ -79,12 +79,190 @@ impl TransformerSqlparser {
...
+                    }
+                    View | Index | Schema => {
+                        return Err(ErrorCode::SyntaxException(format!(
+                            "Unsupported SQL statement: {}",
+                            self.orig_stmt
+                        )));
+                    }
+                }
+            }
+            SqlparserStatement::CreateDatabase {
```
Comment (SGZW) [inline query/src/sql/parser/transformer/transform_sqlparser.rs:242]: create_database is not support for sqlparser parse_statement function

File: query/src/sql/parser/transformer/transform_sqlparser.rs [hunk 2]
```diff
@@ -79,12 +79,190 @@ impl TransformerSqlparser {
...
+                            }
+                            Default(expr) => {
+                                has_default = true;
+                                let res = self.transform_expr(expr);
+                                if res.is_err() {
+                                    return Err(ErrorCode::SyntaxException(format!(
+                                        "Unsupported SQL statement: {}",
+                                        self.orig_stmt
+                                    )));
+                                }
```
Comment (leiysky) [inline query/src/sql/parser/transformer/transform_sqlparser.rs:165]:
```suggestion
                                let res = self.transform_expr(expr)?;
```

Ditto.

File: query/src/sql/parser/transformer/transform_sqlparser.rs [hunk 3]
```diff
@@ -79,12 +79,190 @@ impl TransformerSqlparser {
...
+                columns,
+                ..
+            } => {
+                let (db, tpl) = self.get_database_and_table_from_idents(&name.0);
+                if tpl.is_none() {
+                    return Err(ErrorCode::SyntaxException(format!(
+                        "Unsupported SQL statement: {}",
+                        self.orig_stmt
+                    )));
+                }
```
Comment (leiysky) [inline query/src/sql/parser/transformer/transform_sqlparser.rs:125]: It's better to unwrap `tpl` here like:
```suggestion
                let tpl = tpl.ok_or(Err(ErrorCode::SyntaxException(format!(
                        "Unsupported SQL statement: {}",
                        self.orig_stmt
                    ))))?;
```

So we don't need to handle the `None` case latter.

File: query/src/sql/parser/transformer/transform_sqlparser.rs [hunk 4]
```diff
@@ -79,12 +79,190 @@ impl TransformerSqlparser {
...
+                            return Err(ErrorCode::SyntaxException(format!(
+                                "Unsupported SQL statement: {}",
+                                self.orig_stmt
+                            )));
+                        }
+                        Ok(Statement::DropTable {
+                            if_exists: *if_exists,
+                            database: db,
+                            table: tpl.unwrap(),
+                        })
```
Comment (leiysky) [inline query/src/sql/parser/transformer/transform_sqlparser.rs:232]:
```suggestion
                        if let Some(table) = tpl {
                            Ok(Statement::DropTable {
                                if_exists: *if_exists,
                                database: db,
                                table: tpl.unwrap(),
                            })
                        } else {
                             Err(ErrorCode::SyntaxException(format!(
                                "Unsupported SQL statement: {}",
                                self.orig_stmt
                            )))
                       }
```

Ditto.

**Review by leiysky — COMMENTED** [review]
Rest LGTM

File: query/src/sql/parser/transformer/transform_sqlparser.rs (hunk 1 above)
Comment (SGZW) [reply query/src/sql/parser/transformer/transform_sqlparser.rs:230]: ok, i make a issue for this at sqlparser-rs repo

**Review by leiysky — APPROVED** [review]
LGTM, nice work!

File: query/src/sql/parser/transformer/transform_sqlparser.rs [hunk 5]
```diff
@@ -79,12 +79,178 @@ impl TransformerSqlparser {
...
+    fn get_database_and_table_from_idents(
+        &self,
+        idents: &[Ident],
+    ) -> (Option<Identifier>, Option<Identifier>) {
+        if idents.len() == 1 {
+            (None, Some(Identifier::from(&idents[0])))
+        } else if idents.len() == 2 {
+            (
+                Some(Identifier::from(&idents[0])),
+                Some(Identifier::from(&idents[0])),
```
Comment (sundy-li) [inline query/src/sql/parser/transformer/transform_sqlparser.rs:91]: Seems here should be
`Some(Identifier::from(&idents[1])),`

File: query/src/sql/parser/transformer/transform_sqlparser_test.rs [hunk 6]
```diff
@@ -60,4 +60,41 @@ mod test {
...
         }
     }
+
+    #[test]
+    fn test_parse_with_ddl() {
+        use crate::sql::parser::ast::*;
+        let parser = Parser {};
+        let sqls = vec![
+            "truncate table test",
+            "truncate table test.test",
```
Comment (sundy-li) [inline query/src/sql/parser/transformer/transform_sqlparser_test.rs:70]: set to `truncate table test_db.test_table` can reproduce the error above.

**Pull request merged**